tag.go

```diff
@@ -5,16 +5,25 @@ import "strings"
 type tagValue struct {
 	name     string
 	optional bool
+	flags    map[flag]struct{}
```

Comment:
did you use a map to avoid duplicates?

Reply:
Yes, and to avoid looping to find each flag, just look them up. Not a big deal to switch to a slice if you prefer it.

Reply:
A map is much better!